js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx

```diff
@@ -69,15 +69,22 @@ export const AssetAutomaterializePolicyPage = ({assetKey}: {assetKey: AssetKey})
     return queryResult.data?.autoMaterializeAssetEvaluations || [];
   }, [queryResult.data?.autoMaterializeAssetEvaluations]);
 
-  const [selectedEvaluationId, setSelectedEvaluationId] = useQueryPersistedState({
+  const [selectedEvaluationId, setSelectedEvaluationId] = useQueryPersistedState<
+    number | undefined
+  >({
     queryKey: 'evaluation',
+    decode: (raw) => {
+      try {
+        return raw.evaluation ? parseInt(raw.evaluation) : undefined;
+      } catch (e) {
+        return undefined;
+      }
```

Comment:
Sufficient to do this?

```
const value = parseInt(raw.evaluation);
return isNaN(value) ? undefined : value;
```

Reply:
I think so -- as far as I know this can't throw